src/hotspot/cpu/x86/x86.ad

```diff
@@ -1409,6 +1409,11 @@ const bool Matcher::match_rule_supported(int opcode) {
         return false;
       }
       break;
+    case Op_PopCountVL:
+      if (!UsePopCountInstruction || !VM_Version::supports_avx512_vpopcntdq()) {
+        return false;
+      }
+      break;
```

Comment:
This case could be combined with case Op_PopCountVI and duplication removed. The check is the same for both.

Reply:
Updated code as per your suggestion to avoid duplication in the latest commit...